_posts/2018-06-15-cocoapods-keys-react-native.markdown

```diff
@@ -0,0 +1,156 @@
+[/Pod/Classes/Core/AREmission.m](https://github.com/artsy/emission/blob/4a2a3e9260e97d791536cf38376a06b0ad0946a8/Pod/Classes/Core/AREmission.m#L24-L60):
+
+```diff
+ @implementation AREmissionConfiguration
+ RCT_EXPORT_MODULE(Emission);
+ # ...
+
+ - (NSDictionary *)constantsToExport
+ {
+   return @{
++    @"stripePublishableKey": self.stripePublishableKey ?: @"",
+     # ...lots more
+   };
+ }
+
+ - (instancetype)initWithUserID:(NSString *)userID
+            authenticationToken:(NSString *)token
+                      sentryDSN:(NSString *)sentryDSN
++          stripePublishableKey:(NSString *)stripePublishableKey
+               googleMapsAPIKey:(NSString *)googleAPIKey
+                     gravityURL:(NSString *)gravity
+                 metaphysicsURL:(NSString *)metaphysics
+                      userAgent:(nonnull NSString *)userAgent
+ {
+     self = [super init];
+     _userID = userID.copy;
++    _stripePublishableKey = stripePublishableKey.copy;
+     # ... More copies...
+     return self;
+ }
+```
+
+---
+
+#### 3. Configure the example app to expose the new key to our library
+
+Make sure we have imported they keys from `cocoapods-keys` and initialize the new configuration with the function we defined above.
+
+[Example/Emission/AppDelegate.m](https://github.com/artsy/emission/blob/4a2a3e9260e97d791536cf38376a06b0ad0946a8/Example/Emission/AppDelegate.m#L109)
+
+```objc
+#import <Keys/EmissionKeys.h>
+
+# Add the key to our initWith...  call, following the signature we defined in the previous step
+- (void)setupEmissionWithUserID:(NSString *)userID accessToken:(NSString *)accessToken keychainService:(NSString *)service;
+{
+  # ...
+
+  # stripePublishableKey is somewhere down there...
+
+  AREmissionConfiguration *config = [[AREmissionConfiguration alloc] initWithUserID:userID authenticationToken:accessToken sentryDSN:nil stripePublishableKey:[keys stripePublishableKey] googleMapsAPIKey:nil gravityURL:setup.gravityURL metaphysicsURL:setup.metaphysicsURL userAgent:@"Emission Example"];
+  # ...
+```
+
+---
+
+#### 4. Use that configured key in a `react-native` component.
+
+`Emission` is now exposed along with its configured keys via `react-native`'s `NativeModules`.
+
+```js
+import { NativeModules } from "react-native";
+const Emission = NativeModules.Emission || {};
+
+// ... other setup ...
+
+stripe.setOptions({
+  publishableKey: Emission.stripePublishableKey
+});
+```
+
+<!--
+Cᴏɴᴄʟᴜsɪᴏɴ
+
+The conclusion of your post should mirror the introduction: an overview of the post that describes (but does not explain) the technologies involved. Repetition helps emphasize your points, so make sure to re-state any high-level takeaways from the main body of your post. Maybe a list makes sense, or a paragraph.
+
+The conclusion should also include links to pull requests or issues related to the post, as well as provide any thanks you'd like to extend to your colleagues or community members.
+-->
```

Comment:
Should this be left in?

Reply:
I was just happy to see someone using the blog post templates 😻